Sendspin fixes

maximmaxim345 opened the pull request:
Includes a bunch of other fixes from [aiosendspin](https://github.com/Sendspin/aiosendspin/releases/tag/1.1.1).

maximmaxim345 committed:
Use per-player channel for non-stereo output in Sendspin

Previously, Sendspin only used per-player channels when DSP was enabled.
Now it also uses per-player channels when output channel mode is not
stereo (left, right, or mono), ensuring channel transformations are applied.

maximmaxim345 committed:
Bump aiosendspin

maximmaxim345 commented on music_assistant/providers/sendspin/player.py:
```diff
@@ -533,10 +538,11 @@ async def _send_artist_artwork(self, current_item: QueueItem) -> None:
+        """Extract and send current media metadata to sendspin players."""
         queue = self.mass.player_queues.get_active_queue(self.player_id)
         if not queue or not queue.current_item:
+            self.api.group.set_metadata(Metadata())
```
Its not async.

marcelveldt commented on music_assistant/providers/sendspin/player.py:
```diff
@@ -224,12 +226,8 @@ def __init__(self, provider: SendspinProvider, player_id: str) -> None:
-                (EventType.QUEUE_UPDATED),
-            )
-        )
+        self._attr_needs_poll = True
```
meh why ???

marcelveldt commented on music_assistant/providers/sendspin/player.py:
```diff
@@ -351,6 +349,13 @@ async def group_event_cb(self, group: SendspinGroup, event: GroupEvent) -> None:
+            # Only leader sends metadata
+            return
+        await self._send_metadata()
+
```
I'm not liking the fact that we now are going to poll when that is really unneeded. everything is event driven. So if we are missing an event somewhere, somehow, we should fix that instead.

maximmaxim345 replied:
Agree, removed that. Lets see if it works now (maybe the metadata issues were just in aiosendspin, which I fixed already?).

marcelveldt replied:
Maybe, but we could also be missing one callback in the chain when (only) the player.current_media changes. I will look into that

maximmaxim345 committed:
Fix sendspin metadata updates for grouped players

- Only leader sends metadata (skip when synced_to is set)
- Clear metadata when queue has no current item

maximmaxim345 committed:
Merge branch 'dev' into fix/small-sendspin-fixes
